src/data/write_init_files.py

```diff
@@ -966,11 +966,17 @@ def write_phys_read_subroutine(outfile, host_dict, host_vars, host_imports,
     outfile.write("const_props => cam_model_const_properties()", 7)
     outfile.write("constituent_has_default = .false.", 7)
     outfile.write("call const_props(constituent_idx)%has_default(constituent_has_default, constituent_errflg, constituent_errmsg)", 7)
+    outfile.write("if (constituent_errflg .ne. 0) then", 7)
+    outfile.write("call endrun(constituent_errmsg)", 8)
+    outfile.write("end if", 7)
     outfile.write("if (constituent_has_default) then", 7)
     outfile.write("call const_props(constituent_idx)%default_value(constituent_default_value, constituent_errflg, constituent_errmsg)", 8)
+    outfile.write("if (constituent_errflg .ne. 0) then", 8)
+    outfile.write("call endrun(constituent_errmsg)", 9)
+    outfile.write("end if", 8)
```

Comment:
I think I would make the same modifications here:

```suggestion
    outfile.write("if (constituent_errflg /= 0) then", 8)
    outfile.write("call endrun(constituent_errmsg, file=__FILE__, line=__LINE__)", 9)
    outfile.write("end if", 8)
```